python/GafferSceneUI/HierarchyView.py

```diff
@@ -226,6 +230,18 @@ def __copySelectedPaths( self, *unused ) :
 			data = IECore.StringVectorData( selection.paths() )
 			self.__plug.ancestor( Gaffer.ApplicationRoot ).setClipboardContents( data )
 
+	def __frameLastSelectedPath( self ) :
+
+		selection = self.__pathListing.getSelection()
+		if selection.isEmpty() :
+			return
+
+		self.__pathListing.expandAncestorsOfSelection()
+
+		lastSelectedPath = ContextAlgo.getLastSelectedPath( self.getContext() )
+		if selection.match( lastSelectedPath ) & IECore.PathMatcher.Result.ExactMatch :
+			self.__pathListing.scrollToFirst( IECore.PathMatcher( [ lastSelectedPath ] ) )
```

Comment:
When I first played with this, I was surprised when it didn't scroll to the first selected item in the hierarchy view, but instead to an item lower down. In that case I'd made a marquee selection, so `lastSelectedPath` wasn't really meaningful, but even when I made a multi-selection by individually <kbd>Shift</kbd>+clicking it still felt unnatural to me. 

Perhaps this is just because I've been conditioned to expect the first item from the old auto-scroll-to-selection behaviour, and I'll get used to it? Or perhaps `lastSelectedPath` just isn't very useful in this context?

Reply:
The intent behind using `lastSelectedPath` was to replicate some of the existing behaviour when using shift-click to build up a selection, we would frame based on the new additions to selection rather than the first path found in the entire selection. Though as you point out, `lastSelectedPath` isn't a perfect solution for this and ends up being a bit confusing with marquee selections. For clarity, I've changed the shortcut in 7819e7b to instead scroll to the first path found in the current selection, let me know if that is any more natural feeling to you.

I still think there's value in framing based on the last location(s) selected rather than the current selection, but we'd need to keep track of the latest change to selection rather than the last selected path to do so in a consistent way. Though maybe we can address that as follow-up once 1.3 has had enough use to get some user feedback...

Reply:
> I've changed the shortcut in https://github.com/GafferHQ/gaffer/commit/7819e7b6b6c7aac9401c474f3f81f5ba918602cf to instead scroll to the first path found in the current selection, let me know if that is any more natural feeling to you.

I'm still not certain if my initial reaction was valid or not, but starting with the slightly simpler behaviour and gathering some feedback seems reasonable I think. Will be interesting to see what the consensus is...